### PR TITLE
Ensure properties are always declared in the same order (eslint rule)

### PR DESCRIPTION
**Pull request by Aschen**
## What does this PR do ?

Add the `sort-keys` rule to ensure each object properties is declared in alphabetical order.  
This rule allow to reduce the number of internal hidden classes used to represent objects.

The codebase has been updated with the eslint plugin: https://www.npmjs.com/package/eslint-plugin-sort-keys-fix

:warning: Only the code in `lib/` and `bin/` is affected by the rule and has been modified

### CODE REVIEW

**Comment by Yoann-Abbes** [comment]
It seems like it does not apply the rule inside the units test :(

File: .eslintrc.json [hunk 1]
```diff
@@ -50,8 +50,22 @@
...
+    "yoda": [2, "never"],
+    "sort-keys": ["error", "asc"]
   },
+  "overrides": [
+    {
+      "files": [
+        "test/**/*.js",
+        "features/**/*.js",
+        "features-sdk/**/*.js",
+        "lib/api/controllers/memoryStorage.js"
```
Comment (Aschen) [inline .eslintrc.json]: Memory storage controller use object to declare it's methods, so the property order is important

File: test/core/janitor.test.js [hunk 2]
```diff
@@ -290,8 +290,8 @@ describe('Test: core/janitor', () => {
           .be.exactly(baseDumpPath.concat('/kuzzle.json'));
         should(fsStub.writeFileSync.getCall(0).args[1])
           .be.exactly(JSON.stringify({
-            version: require('../../package.json').version,
-            config: kuzzle.config
+            config: kuzzle.config,
```
Comment (Aschen) [inline test/core/janitor.test.js]: One of the two unit test modified

File: test/core/storage/bootstrap/internalIndexBootstrap.test.js [hunk 3]
```diff
@@ -104,8 +104,8 @@ describe('InternalIndexBootstrap', () => {
       should(internalIndexStorage.createOrReplace.getCall(3).args[0])
         .be.eql('profiles');
       should(ret).be.eql({
-        profileIds: ['admin', 'default', 'anonymous'],
-        roleIds: ['admin', 'default', 'anonymous']
+        profileIds: ['admin', 'anonymous', 'default'],
```
Comment (Aschen) [inline test/core/storage/bootstrap/internalIndexBootstrap.test.js]: One of the two unit test modified

File: lib/config/httpRoutes.js [hunk 4]
```diff
@@ -23,300 +23,300 @@
 
 module.exports = [
   // GET (idempotent)
-  {verb: 'get', url: '/users/_me', controller: 'auth', action: 'getCurrentUser'},
```
Comment (alexandrebouthinon) [inline lib/config/httpRoutes.js]: I think we should make an exception with this file. The  old properties ordering method was more developer friendly IMHO :slightly_smiling_face:

File: lib/util/esWrapper.js [hunk 5]
```diff
@@ -31,97 +31,97 @@ const
 
 const errorMessagesMapping = [
   {
+    getPlaceholders: matches => [matches[1]],
     regex: /^\[es_rejected_execution_exception] rejected execution .*? on EsThreadPoolExecutor\[(.*?), .*$/,
-    subCode: 'too_many_operations',
```
Comment (alexandrebouthinon) [inline lib/util/esWrapper.js]: Same as http routes file

File: .eslintrc.json [hunk 6]
```diff
@@ -50,8 +50,24 @@
...
+    "yoda": [2, "never"],
+    "sort-keys": ["error", "asc"]
   },
+  "overrides": [
+    {
+      "files": [
+        "test/**/*.js",
+        "features/**/*.js",
+        "features-sdk/**/*.js",
+        "lib/api/controllers/memoryStorage.js",
```
Comment (scottinet) [inline .eslintrc.json]: I agree about overriding test files in eslint's configuration file, but I'm not sure about overrides for code files. Chances are we are going to forget about those pretty fast.

I think deactivating the sort-keys rule for these 3 files with a comment directive at the beginning of each file would be safer.

File: lib/config/httpRoutes.js [hunk 7]
```diff
@@ -2,7 +2,7 @@
  * Kuzzle, a backend software, self-hostable and ready to use
  * to power modern apps
  *
- * Copyright 2015-2020 Kuzzle
+ * Copyright 2015-2018 Kuzzle
```
Comment (scottinet) [inline lib/config/httpRoutes.js]: ?

File: lib/config/httpRoutes.js [hunk 8]
```diff
@@ -263,6 +257,8 @@ module.exports = [
   // DELETE
   {verb: 'delete', url: '/credentials/:strategy/_me', controller: 'auth', action: 'deleteMyCredentials'},
 
+  {verb: 'post', url: '/:index/:collection/_refresh', controller: 'collection', action: 'refresh' },
```
Comment (scottinet) [inline lib/config/httpRoutes.js]: should be moved further up, in the "post" section

**Pull request merged**